code/modules/vehicles/mecha/_mecha.dm

```diff
@@ -0,0 +1,1178 @@
+		if(!Adjacent(target) && (selected.range & MECHA_RANGED))
+			if(HAS_TRAIT(L, TRAIT_PACIFISM) && selected.harmful)
+				to_chat(L, "<span class='warning'>You don't want to harm other living beings!</span>")
+				return
+			INVOKE_ASYNC(selected, TYPE_PROC_REF(/obj/item/mecha_parts/mecha_equipment, action), user, target, params)
+			return
+		if((selected.range & MECHA_MELEE) && Adjacent(target))
+			if(isliving(target) && selected.harmful && HAS_TRAIT(L, TRAIT_PACIFISM))
+				to_chat(L, "<span class='warning'>You don't want to harm other living beings!</span>")
+				return
+			INVOKE_ASYNC(selected, TYPE_PROC_REF(/obj/item/mecha_parts/mecha_equipment, action), user, target, params)
+			return
+	if(TIMER_COOLDOWN_CHECK(src, COOLDOWN_MECHA_MELEE_ATTACK) || !istype(target, /atom) || !Adjacent(target))
+		return
+	if(internal_damage & MECHA_INT_CONTROL_LOST)
+		var/list/possible_targets = oview(1,src)
+		if(!length(possible_targets))
+			return
+		target = pick(possible_targets)
+	target.mech_melee_attack(src, user)
+	TIMER_COOLDOWN_START(src, COOLDOWN_MECHA_MELEE_ATTACK, melee_cooldown)
+
+
+//////////////////////////////////
+////////  Movement procs  ////////
+//////////////////////////////////
+
+/obj/vehicle/sealed/mecha/proc/update_step_speed()
+	// Calculate the speed delta
+	// Calculate the move multiplier speed, to be proportional to mob speed
+	// 1.5 was the previous value, so calculate the multiplier in proportion to that
+	step_multiplier = CONFIG_GET(number/movedelay/run_delay) / 1.5
+
+/obj/vehicle/sealed/mecha/proc/play_stepsound()
+	SIGNAL_HANDLER
+	if(mecha_flags & QUIET_STEPS)
+		return
+	playsound(src, stepsound, 40, TRUE)
+
+/obj/vehicle/sealed/mecha/proc/disconnect_air()
+	SIGNAL_HANDLER
+	if(internal_tank.disconnect()) // Something moved us and broke connection
+		to_chat(occupants, "[icon2html(src, occupants)]<span class='warning'>Air port connection has been severed!</span>")
+		log_message("Lost connection to gas port.", LOG_MECHA)
+
+/obj/vehicle/sealed/mecha/Process_Spacemove(movement_dir = 0)
+	. = ..()
+	if(.)
+		return
+
+	var/atom/backup = get_spacemove_backup(movement_dir)
+	if(backup && movement_dir)
+		if(isturf(backup)) //get_spacemove_backup() already checks if a returned turf is solid, so we can just go
+			return TRUE
+		if(istype(backup, /atom/movable))
+			var/atom/movable/movable_backup = backup
+			if((!movable_backup.anchored) && (movable_backup.newtonian_move(turn(movement_dir, 180))))
+				step_silent = TRUE
+				if(return_drivers())
+					to_chat(occupants, "[icon2html(src, occupants)]<span class='info'>The [src] push off [movable_backup] to propel yourself.</span>")
+			return TRUE
+
+	if(active_thrusters?.thrust(movement_dir))
+		step_silent = TRUE
+		return TRUE
+	return FALSE
+
+/obj/vehicle/sealed/mecha/vehicle_move(direction, forcerotate = FALSE)
+	if(!COOLDOWN_FINISHED(src, cooldown_vehicle_move))
+		return FALSE
+	COOLDOWN_START(src, cooldown_vehicle_move, movedelay + step_restricted)
```

Comment:
Should this include the step multiplier?

Reply:
Honestly, I forgot to include it, but I'm not even certain the change is necessary as we are on vehicle_move now instead of snowflake code.

I tested the times across a 34 tile length on box: 
- 11.99 seconds as gygax unaltered
- 05.41 seconds as gygax actuator
- 07.04 on foot

I later tested by swapping between leg actuator and regular gygax mode, and the movespeed changed each time correctly.

The overall goal in #9673 was making them slower than the base carbon movespeed and ensuring they stayed that way, I think that has been achieved.

lmk your thoughts